src/fsspec_xrootd/xrootd.py

```diff
@@ -219,6 +321,163 @@ def ls(self, path: str, detail: bool = True, **kwargs: Any) -> list[Any]:
+                OpenFlags.READ,
+                self.timeout,
+            )
+            if not status.ok:
+                raise OSError(f"File did not open properly: {status.message}")
+
+            max_num_chunks, max_chunk_size = await self._get_max_chunk_info(_myFile)
+            vectors = _make_vectors(chunks, max_num_chunks, max_chunk_size)
+
+            coros = [_async_wrap(_myFile.vector_read, v, self.timeout) for v in vectors]
+
+            results = await _run_coros_in_chunks(
+                coros, batch_size=batch_partition, nofiles=True
+            )
+
+            deets = []
+            for result in results:
+                if not result[0].ok:
+                    raise OSError(
+                        f"File did not vector_read properly: {status.message}"
+                    )
+                for r in result[1]:
+                    deets.append(r.buffer)
+
+        except Exception as exc:
+            raise OSError(f"Something went wrong in _cat_vector_read(): {exc}")
+        finally:
+            status, _n = await _async_wrap(
+                _myFile.close,
+                self.timeout,
+            )
+        return {path: deets}
+
+    async def _cat_ranges(
+        self,
+        paths: list[str],
+        starts: list[int],
+        ends: list[int],
+        max_gap: int | None = None,
+        batch_size: int | None = None,
+        **kwargs: Any,
+    ) -> list[bytes]:
+
+        # TODO: on_error
+        if max_gap is not None:
+            # use utils.merge_offset_ranges
+            raise NotImplementedError
+        if not isinstance(paths, list):
+            raise TypeError
+        if not isinstance(starts, list):
+            raise TypeError
+        if not isinstance(ends, list):
+            raise TypeError
+        if len(starts) != len(paths) or len(ends) != len(paths):
+            raise ValueError
+
+        uniquePaths = defaultdict(list)
+
+        for path, start, end in zip(paths, starts, ends):
+            uniquePaths[path].append((start, end))
+
+        batch_size = batch_size or self.batch_size
+
+        coros = [
+            self._cat_vector_read(key, uniquePaths[key], batch_size)
+            for key in uniquePaths.keys()
+        ]
+
+        results = await _run_coros_in_chunks(coros, batch_size=batch_size, nofiles=True)
+
+        resDict: dict[str, list[bytes]] = defaultdict(list)
+
+        for r in results:
+            resDict.update(r)
+
+        deets = []
+        for path, start, end in zip(paths, starts, ends):
+            d = resDict[path].pop(0)
+            if (end - start) < len(d):
+                a, b = divmod((end - start), len(d))
+                if b > 0:
+                    for _ in range(0, a):
+                        d = d + resDict[path].pop(0)
+                else:
+                    for _ in range(0, a - 1):
+                        d = d + resDict[path].pop(0)
+            deets.append(d)
+
+        return deets
```

Comment:
With the changes I suggested to `_cat_vector_read`, this can be replaced with `return dict(results)`

Reply:
wouldn't that make the function return a dictionary? I thought it's supposed to return a list